Fixed #1715 - Added Functionality to edit role name and description

MYKatz opened the pull request:
Added functionality allowing admins to edit role names and descriptions.  Pressing the edit icons opens up a modal w/ textarea.  Once confirmed, changes to names and descriptions happen in the scope and backend (thru API).  Screenshots:

![screenshot 5](https://user-images.githubusercontent.com/23515048/33799378-38ed3d92-dcdf-11e7-9d16-f3e1739e0ba2.jpeg)
![screenshot 7](https://user-images.githubusercontent.com/23515048/33799379-3d1a3a78-dcdf-11e7-9a7e-57ef072f7c62.jpeg)
![screenshot 8](https://user-images.githubusercontent.com/23515048/33799380-4048536a-dcdf-11e7-9909-49e53a3a7146.jpeg)

mbj36 commented:
Thanks for the PR, It will be reviewed Shortly!

This message was posted by [GitMate.io](https://gitmate.io) with timestamp signature `Thanks for the PR, It will be reviewed Shortly!:1eNm2r:5ZdcjChFTlijArms4bYV_TxPOP4`

MYKatz commented:
The Travis CI build failed due to the PUT method I added in the roleResource.  I'm unsure of how to remedy this while retaining the functionality I added.  A little assistance would be great, thanks!

gkrishnan724 commented on app/views/administration/viewrole.html:
```diff
@@ -47,17 +47,46 @@ <h3 class="bolder">{{'label.heading.enablerole' | translate}}</h3>
     </script>
+    <script type="text/ng-template" id="editName.html">
+		<div class="modal-header silver">
+			<h3 class="bolder">{{'label.edit' | translate}}</h3>
```
Add an `<api-validate></api-validate>` here, since there might be errors when you are editing name. For eg: Try to edit the name to an existing role and permission, the platform will return an error in that case. That error needs to be displayed in the modal.

gkrishnan724 commented on app/views/administration/viewrole.html:
```diff
@@ -47,17 +47,46 @@ <h3 class="bolder">{{'label.heading.enablerole' | translate}}</h3>
                 <tr>
                     <td width="20%">{{'label.heading.name' | translate}}</td>
-                    <td width="80%">{{role.name}}</td>
+                    <td width="80%">{{role.name}} <a class="fa fa-edit" ng-click="showEditName()" has-permission='EDIT_ROLENAME'></a></td>
```
add A `uib-tooltip` here. When you hover over the edit button, you should get a tooltip displaying `Edit Role Name`. Do the same for description too.

MYKatz commented:
@gkrishnan724 Updated 👍

gkrishnan724 commented on app/scripts/services/ResourceFactoryProvider.js:
```diff
@@ -26,7 +26,8 @@
                         disableRoles: {method: 'POST'},
-                        enableRoles: {method: 'POST'}
+                        enableRoles: {method: 'POST'},
+                        put: {method: 'PUT'}
```
Rename this to `update` and then change the code accordingly. Let's see what happens with travis

gkrishnan724 commented on app/views/administration/viewrole.html:
```diff
@@ -47,17 +47,48 @@ <h3 class="bolder">{{'label.heading.enablerole' | translate}}</h3>
                 <tr>
                     <td width="20%">{{'label.heading.name' | translate}}</td>
-                    <td width="80%">{{role.name}}</td>
+                    <td width="80%">{{role.name}} <a class="fa fa-edit" ng-click="showEditName()" has-permission='EDIT_ROLENAME' uib-tooltip="Edit role name"></a></td>
```
The tooltip needs to be labels. sth like `{{label.tooltip.editrolename | translate}}` since these tooltips needs to be translated in different languages too. So please add labels in the `locale-en.json` file

MYKatz commented:
@gkrishnan724 I made the changes you requested.  Unfortunately it seems that TravisCi still rejects this.  Looking at the logs, I think this is because the role resource is hard-coded into the unit test.

gkrishnan724 commented:
Thanks! changes look good

shrunk3 commented:
@MYKatz please look into this, and try to resolve test issues we can merge this

MYKatz commented:
> @MYKatz please look into this, and try to resolve test issues we can merge this

On it. Will be done by mid-April (2024) at the latest.
